src/topology/algebra/monoid.lean

```diff
@@ -4,39 +4,36 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro
 
 Theory of topological monoids.
-
-TODO: generalize `topological_monoid` and `topological_add_monoid` to semigroups, or add a type class
-`topological_operator α (*)`.
 -/
+
 import topology.continuous_on
 import algebra.pi_instances
 
 open classical set filter topological_space
 open_locale classical topological_space big_operators
 
-universes u v w
-variables {α : Type u} {β : Type v} {γ : Type w}
+variables {α : Type*} {β : Type*} {γ : Type*}
 
-section topological_monoid
+/-- Basic hypothesis to talk about a topological additive monoid or a topological additive
+semigroup. A topological additive monoid over α, for example, is obtained by requiring both the
+instances `add_monoid α` and `has_continuous_add α`. -/
+class has_continuous_add (α : Type*) [topological_space α] [has_add α] : Prop :=
+(continuous_add : continuous (λp:α×α, p.1 + p.2))
 
-/-- A topological monoid is a monoid in which the multiplication is continuous as a function
-`α × α → α`. -/
-class topological_monoid (α : Type u) [topological_space α] [monoid α] : Prop :=
+/-- Basic hypothesis to talk about a topological monoid or a topological semigroup.
+A topological monoid over α, for example, is obtained by requiring both the instances `monoid α` and
+`has_continuous_mul α`. -/
+@[to_additive]
```

Comment:
Since you have put the `to_additive` attribute on this definition, you shouldn't need to define also `has_continuous_add` (but you would need to adjust the docstring for the additive version).

Reply:
I thought about this but I did not know it was possible. How do I change the docstring for the additive version if I remove the additive definition?

Reply:
Just `@[to_additive "blah blah"]`.

Reply:
Well if I do remove `has_continuous_add` in lemmas lean complains by saying:
```
unknown declaration 'has_continuous_add.continuous_mul'
```
. Where am I going wrong?

Reply:
It is not figuring out by itself that it also has to change the name of the fields

Reply:
I will change back the tag to request review, as I need someone help, if you want me to change this! `topological_monoid` and other algebraic structure also had a double definition though